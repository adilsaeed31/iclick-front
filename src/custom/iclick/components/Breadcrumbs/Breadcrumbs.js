import React, { Component, Fragment } from "react"
import PropTypes from "prop-types"
import { inject } from "mobx-react"
import ChevronRight from "mdi-material-ui/ChevronRight"
import Link from "components/Link"
import SharedPropTypes from "lib/utils/SharedPropTypes"

@inject("tags")
class Breadcrumbs extends Component {
  static propTypes = {
    isPDP: PropTypes.bool,
    isTagGrid: PropTypes.bool,
    product: PropTypes.object,
    tagId: PropTypes.string,
    tags: PropTypes.arrayOf(SharedPropTypes.tag).isRequired
  }

  renderTagBreadcrumbPiece(tag) {
    const { tags } = this.props

    // Find first tag that is a parent of this tag, if any are
    const parentTag = tags.find(node => node.subTagIds.includes(tag._id))

    return (
      <Fragment>
        {!!parentTag && this.renderTagBreadcrumbPiece(parentTag)}
        <li class="breadcrumb-item">
          <Link route={`/tag/${tag.slug}`}>
            <span>{tag.name}</span>
          </Link>
        </li>
      </Fragment>
    )
  }

  renderTagBreadcrumbs() {
    const { tagId, tags } = this.props

    if (!tagId || !Array.isArray(tags) || tags.length === 0) return null // still loading

    const currentTag = tags.find(tag => tag._id === tagId)
    if (!currentTag) {
      throw new Error(`Unable to find current tag with ID ${tagId}`)
    }

    return this.renderTagBreadcrumbPiece(currentTag)
  }

  renderProductNameBreadcrumb = () => {
    const { product, tagId } = this.props

    if (tagId) {
      return (
        <Fragment>
          {this.renderTagBreadcrumbs()}
          <li class="breadcrumb-item">
            <Link route={`/product/${product.slug}`}>
              <span>{tag.name}</span>
            </Link>
          </li>
        </Fragment>
      )
    }

    return (
      <Fragment>
        <li class="breadcrumb-item">
          <Link route={`/product/${product.slug}`}>
            <span>{product.title}</span>
          </Link>
        </li>
      </Fragment>
    )
  }

  renderBreadcrumbs() {
    const { isPDP, isTagGrid } = this.props

    if (isTagGrid) {
      return this.renderTagBreadcrumbs()
    }

    if (isPDP) {
      return this.renderProductNameBreadcrumb()
    }

    return null
  }

  render() {
    return (
      <nav aria-label="breadcrumb" class="breadcrumb-nav mb-0">
        <div class="container">
          <ol class="breadcrumb">
            <li class="breadcrumb-item">
              <Link route="/">
                <i class="icon-home" />
              </Link>
            </li>
            {this.renderBreadcrumbs()}
          </ol>
        </div>
      </nav>
    )
  }
}

export default Breadcrumbs
