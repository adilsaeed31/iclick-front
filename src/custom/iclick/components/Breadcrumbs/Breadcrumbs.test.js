import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import { Provider } from "mobx-react";
import theme from "custom/reactionTheme";
import Breadcrumbs from "./Breadcrumbs";

const tags = [
  {
    _id: "v2hE7fzL6cJniwgSm",
    createdAt: "2018-06-11T18:00:48.414Z",
    groups: [],
    isDeleted: false,
    isTopLevel: true,
    isVisible: true,
    name: "Tag A",
    shopId: "J8Bhq3uTtdgwZx3rz",
    slug: "tag-a",
    subTagIds: [
      "s3nybDz4QPkmapETc",
      "JB4FRBiWduNsxhhhn",
      "gRM3ADcY77eNJ7Brm"
    ],
    updatedAt: "2018-06-11T18:01:09.568Z"
  }, {
    _id: "s3nybDz4QPkmapETc",
    createdAt: "2018-06-11T18:01:02.411Z",
    groups: [],
    isDeleted: false,
    isTopLevel: false,
    isVisible: true,
    name: "Tag A-1",
    shopId: "J8Bhq3uTtdgwZx3rz",
    slug: "tag-a-1",
    subTagIds: [],
    updatedAt: "2018-06-11T18:01:02.411Z"
  }, {
    _id: "JB4FRBiWduNsxhhhn",
    createdAt: "2018-06-11T18:01:06.975Z",
    groups: [],
    isDeleted: false,
    isTopLevel: false,
    isVisible: true,
    name: "Tag A-2",
    shopId: "J8Bhq3uTtdgwZx3rz",
    slug: "tag-a-2",
    subTagIds: [],
    updatedAt: "2018-06-11T18:01:06.975Z"
  }, {
    _id: "gRM3ADcY77eNJ7Brm",
    createdAt: "2018-06-11T18:01:09.565Z",
    groups: [],
    isDeleted: false,
    isTopLevel: false,
    isVisible: true,
    name: "Tag A-3",
    shopId: "J8Bhq3uTtdgwZx3rz",
    slug: "tag-a-3",
    subTagIds: [],
    updatedAt: "2018-06-11T18:01:09.565Z"
  }
];

const routingStore = {
  tagId: "v2hE7fzL6cJniwgSm"
};

test("tagGrid top level tag", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider primaryShopId={"J8Bhq3uTtdgwZx3rz"} routingStore={routingStore}>
        <Breadcrumbs isTagGrid tagId="v2hE7fzL6cJniwgSm" tags={tags} />
      </Provider>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("tagGrid secondlevel tag", () => {
  const component = renderer.create((
    <MuiThemeProvider theme={theme}>
      <Provider routingStore={routingStore}>
        <Breadcrumbs isTagGrid tagId="s3nybDz4QPkmapETc" tags={tags} />
      </Provider>
    </MuiThemeProvider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
