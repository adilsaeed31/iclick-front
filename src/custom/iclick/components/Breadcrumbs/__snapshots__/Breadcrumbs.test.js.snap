// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tagGrid secondlevel tag 1`] = `
<div
  className="SkBreadcrumbs-container-1"
>
  <a
    className="SkLink-anchor-4"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <span
      className="SkBreadcrumbs-breadcrumbLink-2"
    >
      Home
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 SkBreadcrumbs-breadcrumbIcon-3"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
    />
  </svg>
  <a
    className="SkLink-anchor-4"
    href="/tag/tag-a"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <span
      className="SkBreadcrumbs-breadcrumbLink-2"
    >
      Tag A
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 SkBreadcrumbs-breadcrumbIcon-3"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
    />
  </svg>
  <a
    className="SkLink-anchor-4"
    href="/tag/tag-a-1"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <span
      className="SkBreadcrumbs-breadcrumbLink-2"
    >
      Tag A-1
    </span>
  </a>
</div>
`;

exports[`tagGrid top level tag 1`] = `
<div
  className="SkBreadcrumbs-container-1"
>
  <a
    className="SkLink-anchor-4"
    href="/"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <span
      className="SkBreadcrumbs-breadcrumbLink-2"
    >
      Home
    </span>
  </a>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-5 SkBreadcrumbs-breadcrumbIcon-3"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
    />
  </svg>
  <a
    className="SkLink-anchor-4"
    href="/tag/tag-a"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="link"
    tabIndex={0}
  >
    <span
      className="SkBreadcrumbs-breadcrumbLink-2"
    >
      Tag A
    </span>
  </a>
</div>
`;
